Add mandatory id to rules & Update tests accordingly

The-Huginn opened the pull request:
Resolves #49 

In TriagePullRequestProcessor.java
@Valid annotation was not working in combination with @ConfigFile in method parameters. Maybe due to bug with nested Class. Hence a new method to invoke not null check with @Valid annotated parameter.

xstefank commented on src/main/java/io/xstefank/wildlfy/bot/TriagePullRequestProcessor.java:
```diff
@@ -42,5 +48,13 @@ void onPullRequestOpened(@PullRequest.Opened GHEventPayload.PullRequest pullRequ
+
+    boolean matchRule(GHPullRequest pullRequest, @Valid WildFlyRule rule) {
+        if (Matcher.matches(pullRequest, rule)) {
+            LOG.debug("Matched rule with id: " + rule.id + ".");
```
do you know about debugf?

The-Huginn commented:
I will update this PR soon.

Redoing the logic, to validate ids on startup of the application once.

xstefank commented on src/main/java/io/xstefank/wildlfy/bot/TriagePullRequestProcessor.java:
```diff
@@ -1,21 +1,55 @@
+    @Inject
+    GitHubConfigFileProvider fileProvider;
+
+    @PostConstruct
```
do it with `@Observes StartupEvent` as it is recommended.

xstefank replied:
And put it into a separate class. Don't mix two functionalities in one class.

xstefank commented on src/main/java/io/xstefank/wildlfy/bot/TriagePullRequestProcessor.java:
```diff
@@ -1,21 +1,55 @@
+            for (GHAppInstallation installation : clientProvider.getApplicationClient().getApp().listInstallations()) {
+                GitHub app = clientProvider.getInstallationClient(installation.getId());
+                for (GHRepository repository : app.getInstallation().listRepositories()) {
+                    WildFlyConfigFile wildflyBotConfigFile = fileProvider.fetchConfigFile(repository, "wildfly-bot.yml", ConfigFile.Source.DEFAULT, WildFlyConfigFile.class).get();
```
We should put the name of the config to some shared constant somewhere so it doesn't need to be typed in string everywhere and prone to errors.

The-Huginn replied:
I have opened an issue for this please refer to #53

xstefank commented on src/main/java/io/xstefank/wildlfy/bot/TriagePullRequestProcessor.java:
```diff
@@ -1,21 +1,55 @@
+                    WildFlyConfigFile wildflyBotConfigFile = fileProvider.fetchConfigFile(repository, "wildfly-bot.yml", ConfigFile.Source.DEFAULT, WildFlyConfigFile.class).get();
+                    for (WildFlyRule rule : wildflyBotConfigFile.wildfly.rules) {
+                        if (rule.id == null) {
+                            LOG.errorf("Following rule is missing an id. [%s]", rule);
```
Maybe collect the names and put just one message listing all.

xstefank commented on src/main/java/io/xstefank/wildlfy/bot/TriagePullRequestProcessor.java:
```diff
@@ -28,12 +62,17 @@ void onPullRequestOpened(@PullRequest.Opened GHEventPayload.PullRequest pullRequ
                     }
                 }
+            } else {
+                LOG.warn("Rule is missing an id.");
```
So we need to decide if it's an error or a warning. Do we need to log message on every PR processing? I think that on startup is enough. Well, we should also execute the (current) postconstruct method when a new repo registers. Can you add that please?

The-Huginn replied:
Should I open a new issue for that?

xstefank replied:
No, just remove it here to remove duplications

xstefank commented on src/main/java/io/xstefank/wildlfy/bot/config/WildFlyConfigFile.java:
```diff
@@ -30,5 +32,13 @@ public static final class WildFlyRule {
         @JsonDeserialize(as = TreeSet.class)
         public Set<String> notify = new TreeSet<>();
+
+        private String stringify(String value) {
```
private methods are after public ones

xstefank commented on src/test/java/io/xstefank/wildfly/bot/RuleMissingIdTest.java:
```diff
@@ -0,0 +1,32 @@
+        .when().payloadFromClasspath("/pr-opened.json")
+        .event(GHEvent.PULL_REQUEST)
+        .then().github(mocks -> {
+            verifyNoMoreInteractions(mocks.ghObjects());
```
you might want to try to catch the message log - https://github.com/smallrye/smallrye-health/blob/4c79455b2e78c42a9a745cbabbf7e67c5fde4fd6/implementation/src/test/java/io/smallrye/health/SmallRyeHealthReporterTest.java#L44

The-Huginn commented:
I have addressed most comments.
However I realized, that the test is started with Github AppId in .env file and the initial checks are fired against that app. I have not digged into mocking our own app if even possible. Also the startup event is seemingly fired before the tests could register custom log handler for capturing logs.

xstefank commented on src/main/java/io/xstefank/wildlfy/bot/LifecycleProcessor.java:
```diff
@@ -0,0 +1,49 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public class LifecycleProcessor {
```
I'm surprised this works. Add `@ApplicationScoped` here.

xstefank commented on src/main/java/io/xstefank/wildlfy/bot/config/RuntimeConfiguration.java:
```diff
@@ -0,0 +1,6 @@
+package io.xstefank.wildlfy.bot.config;
+
+public class RuntimeConfiguration {
```
Constants

xstefank commented on src/main/java/io/xstefank/wildlfy/bot/config/RuntimeConstants.java:
```diff
@@ -1,6 +1,6 @@
-public class RuntimeConfiguration {
+public class RuntimeConstants {
 
     public static final String configFile = "wildfly-bot.yml";
```
```suggestion
    public static final String CONFIG_FILE_NAME = "wildfly-bot.yml";
```

The-Huginn commented:
Resolved conflicts with main. Squashed into one commit.

Should be ready to merge.

The-Huginn commented:
Sorry, I didn't restart Quarkus to run wildfly-checkstyle plugin
